mmv1/third_party/terraform/services/bigtable/resource_bigtable_authorized_view_test.go

```diff
@@ -0,0 +1,449 @@
+			},
+		},
+	})
+}
+
+func TestAccBigtableAuthorizedView_destroy(t *testing.T) {
+	// bigtable instance does not use the shared HTTP client, this test creates an instance
+	acctest.SkipIfVcr(t)
+	t.Parallel()
+
+	instanceName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	tableName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	authorizedViewName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+	familyName := fmt.Sprintf("tf-test-%s", acctest.RandString(t, 10))
+
+	acctest.VcrTest(t, resource.TestCase{
+		PreCheck: func() { acctest.AccTestPreCheck(t) },
+
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		CheckDestroy:             testAccCheckBigtableTableDestroyProducer(t),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccBigtableAuthorizedViewWithQualifierPrefixesOnly(instanceName, tableName, authorizedViewName, familyName),
+			},
+			{
+				ResourceName:      "google_bigtable_authorized_view.authorized_view",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccBigtableAuthorizedViewDestroy(instanceName, tableName, familyName),
+			},
+			{
+				ResourceName:      "google_bigtable_table.table",
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func testAccCheckBigtableAuthorizedViewDestroyProducer(t *testing.T) func(s *terraform.State) error {
+	return func(s *terraform.State) error {
+		var ctx = context.Background()
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != "google_bigtable_authorized_view" {
+				continue
+			}
+
+			config := acctest.GoogleProviderConfig(t)
+			c, err := config.BigTableClientFactory(config.UserAgent).NewAdminClient(config.Project, rs.Primary.Attributes["instance_name"])
+			if err != nil {
+				// The instance is already gone
+				return nil
+			}
+
+			_, err = c.AuthorizedViewInfo(ctx, rs.Primary.Attributes["table_name"], rs.Primary.Attributes["name"])
+			if err == nil {
+				return fmt.Errorf("AuthorizedView still present. Found %s in %s.", rs.Primary.Attributes["name"], rs.Primary.Attributes["table_name"])
+			}
+
+			c.Close()
+		}
+
+		return nil
+	}
+}
+
+func testAccBigtableAuthorizedViewInvalidDeletionProtection(instanceName, tableName, authorizedViewName, familyName string) string {
+	return fmt.Sprintf(`
+resource "google_bigtable_instance" "instance" {
+  name          = "%s"
+  cluster {
+    cluster_id = "%s"
+    zone       = "us-central1-b"
+	num_nodes  = 1
+  }
+  deletion_protection = false
+}
+
+resource "google_bigtable_table" "table" {
+  name          = "%s"
+  instance_name = google_bigtable_instance.instance.id
+  column_family {
+	  family = "%s"
+  }
+}
+resource "google_bigtable_authorized_view" "authorized_view" {
+  name         = "%s"
+  instance_name = google_bigtable_instance.instance.id
+  table_name = google_bigtable_table.table.name
+  deletion_protection = "random"
+  subset_view {}
```

Comment:
What is the difference between setting `subset_view {}` and not setting `subset_view` at all?

Reply:
It's related to the comment above.

Setting `subset_view {}` makes sure that the one-of field in the API proto is set to subset_view variant (although it's an empty subset_view). Otherwise the one-of field in the API proto is not set at all.

Reply:
Thank you for the explanation